SAK-31460 Changes for optional Quick Links functionality.

RebeccaOxfordUni opened the pull request.

buckett commented:
Build fails and looks related to this change.

RebeccaOxfordUni commented:
Minor change made.

buckett commented on reference/library/src/morpheus-master/sass/tool.scss:
```diff
@@ -44,6 +44,7 @@
 @import "base/rtl";
 
 @import "modules/access/base";
+@import "modules/quicklinks/_base";
```
I think this should be `/base` to keep it the same as the surrounding lines.

jonespm commented:
Added some comments to the jira

ottenhoff commented:
Should we keep this open?

buckett commented:
I think @RebeccaOxfordUni is going to look at sorting this out.

RebeccaOxfordUni commented:
I've added some comments in the Jira.

RebeccaOxfordUni commented:
Hopefully fixed the issue that was breaking the build.

jonespm commented:
It looks like this is missing some commits.
1) I get an error in the UI missing key (mre): sitenav sit_quick_links]
  - I don't see any message bundle additions committed here
2) Why is it called 'related links' in the code rather than 'quick links' this should be consistent
3) Property should be prefixed with portal. and also probably called quicklink too rather than related.link?
  - portal.quicklink.url=
4) I'd remove the oxford from the property example since the image isn't there and the example doesn't work. Also using new style propreties is better (This would be portal.quicklink.url)
```
related.link.url=https://sakaiproject.org/,https://www.apereo.org
related.link.title=Sakai Project Website,Apereo Website
related.link.name=Sakai Project,Apereo
related.link.icon=im /library/image/sakai/url.gif,cl fa fa-glob
```

jonespm commented:
It also doesn't seem to be hiding the relatedLinks div to me? Maybe this is just some mixup with some parts being quickLinks and some being relatedLinks? 

![image](https://cloud.githubusercontent.com/assets/27447/19048127/de91b47e-8959-11e6-8724-253375691002.png)

jonespm commented on portal/portal-render-engine-impl/pack/src/webapp/vm/morpheus/includeLoginNav.vm:
```diff
@@ -3,11 +3,51 @@
 <nav id="mastLogin" class="Mrphs-loginNav">
 
     #if (!${loginTopLogin})
+        <div id="relatedLinks">
```
This div seems unnecessary with the quickLinks div below?

RebeccaOxfordUni commented:
Thanks for the feedback Matthew.  I've added the sitenav properties and changed the property names to be the new style you mention above.  The code now calls everything quick links instead of related links.

The extra div you mentioned has been removed which should sort out the quicklinks showing when they shouldn't.  I'll review cherry-picking from our Oxford code more thoroughly next time!

Thanks once again for looking at this.

jonespm commented:
I tested this out and think it works good, keyboard focus works too which is good for a11y. Thanks for the changes!
